hermes-management/src/main/java/pl/allegro/tech/hermes/management/domain/topic/TopicOwnerCache.java

```diff
@@ -22,20 +27,31 @@
 
     private final TopicRepository topicRepository;
     private final GroupService groupService;
+    private final ScheduledExecutorService scheduledExecutorService;
 
     private final Multimap<OwnerId, TopicName> cache = Multimaps.synchronizedMultimap(ArrayListMultimap.create());
 
-    public TopicOwnerCache(TopicRepository topicRepository, GroupService groupService) {
+    public TopicOwnerCache(TopicRepository topicRepository, GroupService groupService,
+                           @Value("${topicOwnerCache.refreshRateInSeconds}") int refreshRateInSeconds) {
         this.topicRepository = topicRepository;
         this.groupService = groupService;
+        scheduledExecutorService = Executors.newSingleThreadScheduledExecutor(
+                new ThreadFactoryBuilder()
+                        .setNameFormat("topic-owner-cache-%d")
+                        .build());
+        scheduledExecutorService.scheduleAtFixedRate(this::fillCache, 0, refreshRateInSeconds, TimeUnit.SECONDS);
+    }
+
+    @PreDestroy
+    public void stop() {
+        scheduledExecutorService.shutdown();
     }
 
     public Collection<TopicName> get(OwnerId ownerId) {
         return cache.get(ownerId);
     }
 
-    @PostConstruct
-    public void fillCache() {
+    private void fillCache() {
```

Comment:
może dajmy tutaj try-catch prewencyjnie, skoro to zaciąga coś z zookeepera to może wylecieć jakiś wyjątek i executor się zatrzyma